scene/3d/ray_cast_3d.cpp

```diff
@@ -353,26 +456,39 @@ void RayCast3D::_update_debug_shape() {
 		return;
 	}
 
-	Vector<Vector3> verts;
-	verts.push_back(Vector3());
-	verts.push_back(target_position);
+	_update_debug_shape_vertices();
 
 	if (mesh->get_surface_count() == 0) {
```

Comment:
I didn't think about it before, but the mesh surface creation would be much safer this way, in order to avoid problems when switching between no line at all (length 0), just line (thickness 1) and line + shape (thickness > 1):
(it just clears and creates the surfaces again every time instead of changing the vertices)
```
	_update_debug_shape_vertices();

	mesh->clear_surfaces();

	Array a;
	a.resize(Mesh::ARRAY_MAX);

	uint32_t flags = 0;
	int surface_count = 0;

	if (!debug_line_vertices.is_empty()) {
		a[Mesh::ARRAY_VERTEX] = debug_line_vertices;
		mesh->add_surface_from_arrays(Mesh::PRIMITIVE_LINES, a, Array(), Dictionary(), flags);
		mesh->surface_set_material(surface_count, debug_material);
		++surface_count;
	}

	if (!debug_shape_vertices.is_empty()) {
		a[Mesh::ARRAY_VERTEX] = debug_shape_vertices;
		mesh->add_surface_from_arrays(Mesh::PRIMITIVE_TRIANGLE_STRIP, a, Array(), Dictionary(), flags);
		mesh->surface_set_material(surface_count, debug_material);
		++surface_count;
	}
```